pkg/odo/cli/component/create.go

```diff
@@ -441,11 +502,12 @@ func NewCmdCreate(name, fullName string) *cobra.Command {
 	componentCreateCmd.Flags().StringVar(&co.cpuMax, "max-cpu", "", "Limit maximum amount of cpu to be allocated to the component. ex. 1")
 	componentCreateCmd.Flags().StringSliceVarP(&co.componentPorts, "port", "p", []string{}, "Ports to be used when the component is created (ex. 8080,8100/tcp,9100/udp)")
 	componentCreateCmd.Flags().StringSliceVar(&co.componentEnvVars, "env", []string{}, "Environmental variables for the component. For example --env VariableName=Value")
-
 	// Add a defined annotation in order to appear in the help menu
 	componentCreateCmd.Annotations = map[string]string{"command": "component"}
 	componentCreateCmd.SetUsageTemplate(odoutil.CmdUsageTemplate)
 
+	// Adding `--now` flag
+	genericclioptions.AddNowFlag(componentCreateCmd, &co.now)
```

Comment:
I don't really understand why we are resorting to using a function to add a parameter, using:
```sh
componentCreateCmd.Flags().StringSliceVar(&co.componentEnvVars, "env", []string{}, "Environmental variables for the component. For example --env VariableName=Value")
```

(similar to that)

Already sets a default value if nothing is provided.. I suggest we stick to the conventional way of how we're adding flags or else this may seem like over-engineering the `cli` package.. That's just my opinion though.

Reply:
Got that this is for consolidating flags that might be used across more than a couple of locations, which this flag is.